services/alerts/src/AlertsManagerContext.ts

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import { Alert, AlertsManager, AlertsManagerAlert } from './types'
+
+type SetAlertsFunction = React.Dispatch<React.SetStateAction<Alert[]>>
+
+export const makeAlertManager = (
+    setAlerts: SetAlertsFunction
+): AlertsManager => {
+    let id = 0
+
+    const remove = (alert: Alert) => {
```

Comment:
Renamed this from `hide` to `remove` to better reflect the fact that the `alertsManager` just keeps track of an array of alerts. Showing and hiding is up to the UI.

Reply:
Sounds good

Reply:
Actually (unrelated)... I think we should update this to take an `id` rather than an `Alert` object reference

Reply:
I think this makes sense, but do want to emphasise one thing: this is something we could do for the `remove` function, **but not for `add`**. The reason is this: https://github.com/dhis2/app-runtime/blob/c93d8cf4873ae5dc34d47fa8bb51f2fd320bd63a/services/alerts/src/AlertsManagerContext.ts#L27-L31

So we only increment the ID when we find an alert that is actually new, i.e. a new call to `useAlert`. Perhaps there are ways around this, but I think that would just complicate the implementation.

Perhaps this was already obvious to you both but I just wanted to make sure.

Reply:
See the other discussion, I think this check is superfluous. If for some reason we need this check, I'd like to delve deeper into this, because then this would seem to hint at an issue with the concept

Reply:
After staring at the code above for a while I realised that this part of the code can't really be doing anything anyway. It’s checking if the `alert` that is received as arg is already in the `alerts` array, but that can never happen, because a few lines down we declare a new object and that’s what we actually add to the alerts array. So the by-reference check `alerts.some(a => a === alert)` will return `false` 100% of the time.

So, I've removed this code for now. If later on it turns out we do need to prevent duplicate alerts from being displayed, we need to implement this differently, probably by using a deep/shallow equality check.